### PR TITLE
Add tests for store.wait()

### LINKED ISSUE #53863: Add tests for store.wait()
There are no test cases for [store.wait()](https://pytorch.org/docs/master/distributed.html#torch.distributed.Store.wait). We should add them for TCPStore, FileStore, and HashStore in https://github.com/pytorch/pytorch/blob/master/test/distributed/test_store.py

## How to get started

0.1. (optional) get conda and create conda env
1. install dependences - https://github.com/pytorch/pytorch#install-dependencies
2. get the source - https://github.com/pytorch/pytorch#get-the-pytorch-source
3. install - https://github.com/pytorch/pytorch#install-pytorch
4. Once torch is installed with no errors. Test `python -c "import torch; print(torch.__version__)"` to see you have the correct version, it should look something like: 2.0.0a0+git54020ba
5. Create your own python script and try to call the APIs you are testing. Note if you make only python changes to pytorch, you do NOT need to rebuild.

## How to run a test

1. `pip install pytest`
2. Use this test as an example `test_address_already_in_use` (https://github.com/pytorch/pytorch/blob/master/test/distributed/test_store.py#L205)
3. `pytest test/distributed/test_store.py -vsk test_address_already_in_use`
4. Check out https://github.com/pytorch/pytorch/blob/master/torch/distributed/CONTRIBUTING.md#running-unit-tests for more details

cc @gravityknife

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang @kwen2501 @agolynski @mrzzd @cbalioglu

### PR DESCRIPTION
**Pull request by gravityknife**
Fixes #53863

Test command 
`pytest test/distributed/test_store.py -vsk test_wait`

### CODE REVIEW

File: test/distributed/test_store.py [hunk 1]
```diff
@@ -110,6 +110,18 @@ def _test_compare_set(self, store):
     def test_compare_set(self):
         self._test_compare_set(self._create_store())
 
+    def assertRaisesRegex(self, i, j):
```
Comment (H-Huang) [inline test/distributed/test_store.py:113]: i think there should be an python unittest function, so you dont need to define it yourself

Comment (gravityknife) [reply test/distributed/test_store.py]: Done, thanks

File: test/distributed/test_store.py [hunk 2]
```diff
@@ -110,6 +110,18 @@ def _test_compare_set(self, store):
...
+        raise RuntimeError("not implemented")
+
+    def _test_wait(self, fs):
+        with self.assertRaisesRegex(RuntimeError, "[t -i]imeout"):
+            fs.wait(["bad_key"], timedelta(seconds=0.25))
+        fs.add("good_key", 1)
+        fs.wait(["good_key"])
```
[the remark this answers is not in the document]
Comment (gravityknife) [reply test/distributed/test_store.py:119]: Done.

**Pull request closed without merging**